### ISSUE TITLE
Set.prototype.{ flatMap, flat }

### ISSUE THREAD

**Issue by zloirock** [tc39/proposal-collection-methods#34]
So, `flatMap` and `flat` are added to `Array` in ES2019. In which form they should be added to this proposal? What they should flatten - sets, arrays, iterables?

**Comment by zloirock** [comment]
Makes sense to see iterators `.flatMap` semantic https://tc39.es/proposal-iterator-helpers/#sec-iteratorprototype-flatmap.

**Comment by Ginden** [comment]
I'm not sure if there is any intuitive semantics for `Map.prototype.flatMap` and `Map.prototype.flat`. `Set.prototype.flatMap` is quite obvious, I will add it.
